src/meta-srv/src/procedure/region_failover/invalidate_cache.rs

```diff
@@ -0,0 +1,349 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::Duration;
+
+use api::v1::meta::MailboxMessage;
+use async_trait::async_trait;
+use common_meta::instruction::{Instruction, InstructionReply, SimpleReply, TableIdent};
+use common_meta::RegionIdent;
+use common_telemetry::{debug, warn};
+use futures::future::join_all;
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+
+use super::failover_end::RegionFailoverEnd;
+use super::{RegionFailoverContext, State, INVALIDATE_TABLE_CACHE_TIMEOUT};
+use crate::error::{self, Result};
+use crate::handler::HeartbeatMailbox;
+use crate::service::mailbox::{Channel, MailboxReceiver};
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub(super) struct InvalidateCache;
+
+impl InvalidateCache {
+    async fn send_invalidate_table_cache_message(
+        &self,
+        ctx: &RegionFailoverContext,
+        receivers: Vec<Channel>,
+        table_ident: &TableIdent,
+        timeout: Duration,
+    ) -> Result<Vec<Result<MailboxReceiver>>> {
+        let instruction = Instruction::InvalidateTableCache(table_ident.clone());
+
+        let futures = receivers
+            .iter()
+            .map(|receiver| {
+                let msg = MailboxMessage::json_message(
+                    "Invalidate Table Cache",
+                    &format!("Metasrv@{}", ctx.selector_ctx.server_addr),
+                    &format!("{receiver}"),
+                    common_time::util::current_time_millis(),
+                    &instruction,
+                )
+                .with_context(|_| error::SerializeToJsonSnafu {
+                    input: instruction.to_string(),
+                })?;
+
+                Ok(ctx.mailbox.send(receiver, msg, timeout))
+            })
+            .collect::<Result<Vec<_>>>()?;
+
+        Ok(join_all(futures).await)
+    }
+
+    async fn handle_mailbox_reply(
+        mailbox_receiver: MailboxReceiver,
+        table_ident: &TableIdent,
+    ) -> Result<()> {
+        let ch = mailbox_receiver.channel();
+        match mailbox_receiver.await? {
+            Ok(msg) => {
+                debug!("Received invalidate table cache reply: {msg:?}");
+                let reply = HeartbeatMailbox::json_reply(&msg)?;
+                let InstructionReply::InvalidateTableCache(SimpleReply { result, error }) = reply else {
+                    return error::UnexpectedInstructionReplySnafu {
+                        mailbox_message: msg.to_string(),
+                        reason: "expect invalidate table cache reply"
+                    }.fail();
+                };
+
+                if result {
+                    Ok(())
+                } else {
+                    let reason = format!(
+                        "table cache {table_ident:?} is not invalidated by Frontend {}, error: {error:?}",
+                        ch,
+                    );
+                    error::RetryLaterSnafu { reason }.fail()
```

Comment:
I don't think we need to retry here. Table routes in Frontend nodes are just cache, they can be updated periodically or upon some failures. Here adding the step of invalidate table cache in Frontend in region failover procedure is merely an optimization to client reads/writes, not a necessity. 

Even more, I don't think Meta should wait for Frontend's reply, just "fire and forget".